src/System.Management.Automation/engine/hostifaces/InternalHostUserInterface.cs

```diff
@@ -257,7 +265,7 @@ public override
                 return;
             }
 
-            _externalUI.WriteLine(value);
+            _externalUI.WriteLine(Utils.GetOutputString(value, isHost: true, SupportsVirtualTerminal, Console.IsOutputRedirected));
```

Comment:
I'm also concerned about the check on `Console.IsOutputRedirected`. For the .NET interactive PS sub kernel, the .NET interactive process's standard output is redirected, so the output can be send to the frontend, but we don't want to lose the ANSI sequences in that case becasue most likely the frontend will support ANSI sequences, such as the jupyter notebook.

I think, maybe the call to `Utils.GetOutputString` should be moved to a lower layer, for example, in PowerShell console host, move to `ConsoleHostUserInterface`, and let the implementation of the `_externalUI` to decide whether the console output/error is considered redirected.

Reply:
So what I mean in general is, for an application that hosts PowerShell, `Console.IsOutputRedirected` may not make sense at all, for example, the hosting application is a GUI application that don't have a console assigned to it. So, it may make more sense if the decision is made at the `_externalUI` level instead.

Reply:
With the move of the code out of `InternalHostInterface.cs` and into `ConsoleHostUserInterface.cs` there isn't any check of redirection outside of consolehost so this is resolved.